src/Microsoft.AspNetCore.Mvc.Core/RequireHttpsAttribute.cs

```diff
@@ -82,6 +88,9 @@ protected virtual void HandleNonHttpsRequest(AuthorizationFilterContext filterCo
                     host = new HostString(host.Host);
                 }
 
+                //i use MvcOption.requireHttpsPermanent value if Permanent parameter is null
+                _permanent = _permanent ?? optionsAccessor.Value.RequireHttpsPermanent;
```

Comment:
Just thought about one thing... Should this be assigned back to the field, or should we keep falling back to `optionsAccessor.Value.RequireHttpsPermanent` for each call? Can the option change?